<launch>

  <!-- start load driver nodelet -->
  <arg name="model" default="RS16" />
  <arg name="ip" default=""/>
  <arg name="msop_data_port" default="6699"/>
  <arg name="difop_data_port" default="7788"/>
  <arg name="rpm" default="600.0" />
  <arg name="frame_id" default="rslidar" />
  <arg name="topic" default="/apollo/sensor/rslidar/rslidarScan" />
  <arg name="node_name" default="driver_nodelet"/>
  <arg name="nodelet_manager_name" default="rslidar_nodelet_manager" />

  <node pkg="nodelet" type="nodelet" name="$(arg node_name)"
        args="load rslidar_driver/DriverNodelet $(arg nodelet_manager_name)" output="screen" >
    <param name="model" value="$(arg model)"/>
    <param name="rpm" value="$(arg rpm)"/>
    <param name="frame_id" value="$(arg frame_id)"/>
    <param name="topic" value="$(arg topic)"/>    
    <param name="ip" value="$(arg ip)"/>
    <param name="msop_data_port" value="$(arg msop_data_port)"/>
    <param name="difop_data_port" value="$(arg difop_data_port)"/>
  </node>    

</launch>
